vllm/ex/ex.py

```diff
@@ -0,0 +1,245 @@
+import copy
+import torch
+import unittest.mock
+
+from .code_cache import CodeCache
+from .fusion import FusedOpGenerator, pointwise_fusion
+from .register import SUPPORTED
+from .rewrite_quantized_gemms import rewrite_quantized_gemms
+from .utils import ModuleInputGenerator, graph_print_tabular, is_call, call_method_class
+
+from torch._dynamo import register_backend, lookup_backend
+from torch.fx.passes.operator_support import create_op_support
+from torch.fx.passes.infra.partitioner import CapabilityBasedPartitioner, Partition
+from torch.fx.passes.tools_common import get_node_target
+from torch.fx.passes.shape_prop import ShapeProp
+from torch._subclasses.fake_tensor import FakeTensorMode, FakeTensor
+
+from typing import List, Tuple, Any, Dict, Optional, Callable, Mapping, Set
+
+from vllm.logger import init_logger
+
+import traceback
+
+logger = init_logger(__name__)
+
+###############################################################################
+#
+# Module partitioning
+#
+###############################################################################
+
+"""
+A callback for the fx CapabilityBasedPartitioner.  Nodes that are "supported"
+are partitioned into new submodules.
```

Comment:
I think extending this comment to explain why both a node and its target need to be in the support list of operations would be helpful here. I assume it's because you are going to pass in `allows_single_node_partition=False,` to the partitioner in `partition_graph`?

Reply:
The target is what pytorch calls the function/method/etc. that's contained in the fx.node.  It doesn't have anything to do with successors in the graph or users of the output of the node.